src/ORM/Association.php

```diff
@@ -436,13 +436,31 @@ public function property($name = null)
             $this->_propertyName = $name;
         }
         if ($name === null && !$this->_propertyName) {
-            list(, $name) = pluginSplit($this->_name);
-            $this->_propertyName = Inflector::underscore($name);
+            $this->_propertyName = $this->_propertyName();
+            if (in_array($this->_propertyName, $this->_sourceTable->schema()->columns())) {
+                $msg = 'Association property name "%s" clashes with field of same name of table "%s".' .
+                    "\n" . 'You should explicitly specify the "propertyName" option.';
```

Comment:
I'm not sure a newline is necessary. But I don't have strong convictions on that.